### PR TITLE
Support multiple supervisde evals, some cleanup around that

### PR DESCRIPTION
**Pull request by dlwh**

### CODE REVIEW

File: src/levanter/data/sharded_datasource.py [hunk 1]
```diff
@@ -184,31 +184,6 @@ def gcs_glob(pattern: str) -> list[str]:
     return matching_urls
 
 
-def datasource_from_chat_jsonl(
```
Comment (dlwh) [inline src/levanter/data/sharded_datasource.py:187]: this just moved to text

File: src/levanter/data/text.py [hunk 2]
```diff
@@ -722,28 +796,69 @@ def _prepare_supervised_example(ex: dict, tokenizer: PreTrainedTokenizerBase, Po
...
+
+                tags = [name] + tags
+            else:
+                name = "default"
+            out[name] = (ds, tags)
+
+    return out
+
+
+def mk_supervised_dataset(
```
Comment (TheQuantumFractal) [inline src/levanter/data/text.py:852]: Seems like we are replacing this with mk_supervised_datasets? Do we still need this function then?

Comment (dlwh) [reply src/levanter/data/text.py:852]: it's still used in sft.py atm but i'd like to remove it

**Review by ahmeda14960 — APPROVED** [review]
Overall LGTM!  I think it's probably better that we get this merged into main, and I can change the revise sft PR accordingly.

File: src/levanter/data/text.py (hunk 2 above)
Comment (TheQuantumFractal) [reply src/levanter/data/text.py:852]: Sounds good

**Pull request merged**